packages/app/src/components/inspector/SymbolInstanceOverridesRow.tsx

```diff
@@ -0,0 +1,314 @@
+import { InputField, Spacer, TreeView, Select } from 'noya-designsystem';
+import { ApplicationState, Layers, Overrides } from 'noya-state';
+import {
+  getSharedStyles,
+  getSharedTextStyles,
+  getSymbols,
+} from 'noya-state/src/selectors/themeSelectors';
+import { memo, ReactNode, useCallback, useMemo } from 'react';
+import { useApplicationState } from '../../contexts/ApplicationStateContext';
+import * as InspectorPrimitives from './InspectorPrimitives';
+
+const NO_STYLE = '';
+
+interface Props {
+  overrideValues: Sketch.OverrideValue[];
+  symbolMaster: Sketch.SymbolMaster;
+  onSetOverrideValue: (overrideName: string, value: string) => void;
+}
+
+type Selectors = {
+  symbols: Sketch.SymbolMaster[];
+  textStyles: Sketch.SharedStyle[];
+  themeStyles: Sketch.SharedStyle[];
+};
+
+const ThemeStyleSelector = ({
+  themeStyles,
+  sharedStyleID,
+  onChange,
+}: {
+  themeStyles: Sketch.SharedStyle[];
+  sharedStyleID: string;
+  onChange: (value: string) => void;
+}) => {
+  const themeStylesOptions = useMemo(
+    () => [NO_STYLE, ...themeStyles.map((style) => style.do_objectID)],
+    [themeStyles],
+  );
+
+  const getThemeStylesTitle = useCallback(
+    (id) =>
+      id === NO_STYLE
+        ? 'No Layer Style'
+        : themeStyles.find((style) => style.do_objectID === id)!.name,
+    [themeStyles],
+  );
+
+  return (
+    <Select
+      id="theme-style-selector"
+      value={sharedStyleID}
+      options={themeStylesOptions}
+      getTitle={getThemeStylesTitle}
+      onChange={onChange}
+    />
+  );
+};
+
+const TextStyleSelector = ({
+  textStyles,
+  sharedStyleID,
+  onChange,
+}: {
+  textStyles: Sketch.SharedStyle[];
+  sharedStyleID: string;
+  onChange: (value: string) => void;
+}) => {
+  const textStylesOptions = useMemo(
+    () => [NO_STYLE, ...textStyles.map((style) => style.do_objectID)],
+    [textStyles],
+  );
+
+  const getTextStylesTitle = useCallback(
+    (id) =>
+      id === NO_STYLE
+        ? 'No Layer Style'
+        : textStyles.find((style) => style.do_objectID === id)!.name,
+    [textStyles],
+  );
+  return (
+    <Select
+      id="text-style-selector"
+      value={sharedStyleID}
+      options={textStylesOptions}
+      getTitle={getTextStylesTitle}
+      onChange={onChange}
+    />
+  );
+};
+
+const SymbolMasterSelector = ({
+  symbols,
+  symbolId,
+  onChange,
+}: {
+  symbols: Sketch.SymbolMaster[];
+  symbolId: string;
+  onChange: (value: string) => void;
+}) => {
+  const symbolSourceOptions = useMemo(
+    () => symbols.map((symbol) => symbol.symbolID),
+    [symbols],
+  );
+
+  const getSymbolMasterTitle = useCallback(
+    (id) => symbols.find((symbol) => symbol.symbolID === id)!.name,
+    [symbols],
+  );
+  return (
+    <Select
+      id="symbol-instance-source"
+      value={symbolId}
+      options={symbolSourceOptions}
+      getTitle={getSymbolMasterTitle}
+      onChange={onChange}
+    />
+  );
+};
+
+function getOverrideElements(
```

Comment:
I wonder if there's some aspect of this we can share with the SymbolMasterInspector 🤔

Reply:
I'm checking this 🤔